jenkins/ci.suse.de/openstack-mkcloud.yaml

```diff
@@ -428,6 +428,8 @@
             export automationrepo=$WORKSPACE/automation-git
           fi
           exec ${automationrepo}/scripts/jenkins/mkcloud/openstack-mkcloud.builder.sh
+          # subunit2junitxml will fail if test run failed as it forwards subunit stream result code, ignore it
+          subunit2junitxml .artifacts/tempest.subunit.log > .artifacts/results.xml || true
```

Comment:
1.
This code will never be run.
You should rather create a new build step.

2.
Where does the `subunit2junitxml` come from on the build node (keep in mind, all our jobs should be self-bootstrapping and caring about its dependencies, otherwise make sure to add requirements to the cloudsalt repo)

3.
The `|| true` hides errors in the run of `subunit2junitxml`. Rather check for the file and and only run the command if its there.

Reply:
1. Why is not going to run after executing mkcloud?
2. I just did the question in the commit message because I'm not familiar with the environment and I was waiting for somebody to offer more guidance, will create a pullrequest to cloudsalt repo, thx.
3. It's commented in the code,  it's the only way to make it work, otherwise, the failing tests make the job to fail. I know that's not the best and elegant thing but you can see lot of Openstack upstraem examples with this approach. Also, you can check in this [link](https://ci.opensuse.org/view/OpenStack/job/openstack-devstack-ipv4/48/) that Jenkins differentiate between a job failure and a test failure

Reply:
Nothing after an `exec` will run.

Reply:
@jdsn I'm to familiar with cloudsalt, could you let me know if it is possible and how can we add  subuni2junitxml and junitxml to the jenkins slave nodes?

@JanZerebecki how can  I run the subunit2junitxml after the `exec`?

Reply:
Remove the exec from the front of that line.